kartothek/io/eager.py

```diff
@@ -236,7 +240,6 @@ def read_table(
     store=None,
     table=None,
     columns=None,
-    concat_partitions_on_primary_index=False,
```

Comment:
No, don't remove it. Please add a deprecation warning and keep it for now. We'll remove this with the next major release